iceoryx_posh/include/iceoryx_posh/popo/modern_api/base_subscriber.hpp

```diff
@@ -31,12 +31,16 @@ using uid_t = UniquePortId;
 template <typename T, typename port_t = iox::SubscriberPortUserType>
 class BaseSubscriber : public Condition
 {
-  public:
+  protected:
+    BaseSubscriber(const capro::ServiceDescription& service);
     BaseSubscriber(const BaseSubscriber& other) = delete;
     BaseSubscriber& operator=(const BaseSubscriber&) = delete;
     BaseSubscriber(BaseSubscriber&& rhs) = delete;
     BaseSubscriber& operator=(BaseSubscriber&& rhs) = delete;
-    ~BaseSubscriber();
+    virtual ~BaseSubscriber();
+
+    // Required for testing. Allows BaseSubscriber instance to be constructed without a PoshRuntime.
+    BaseSubscriber() noexcept {};
```

Comment:
If it's only used for testing, please make this protected and create in the test something like this
```
class BaseSubscriberSUT : public BaseSubscriber {};
```
Now you can create a `BaseSubcriverSUT`, which basically is a `BaseSubscriber` without a `PoshRuntime` and you still have the benefit that it's not accidentally possible in the remaining code base.
Btw. this can also be used to make protected method accessible in tests, just add a `public: using methodName;` to the SUT class.